src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotDeleteSamGroupStep.java

```diff
@@ -27,26 +27,33 @@ public class DeleteSnapshotDeleteSamGroupStep extends DefaultUndoStep {
    * @param iamService
    * @param snapshotId
    */
-  public DeleteSnapshotDeleteSamGroupStep(IamService iamService, UUID snapshotId) {
+  public DeleteSnapshotDeleteSamGroupStep(
+      IamService iamService, SnapshotRequestDao snapshotRequestDao, UUID snapshotId) {
     this.iamService = iamService;
+    this.snapshotRequestDao = snapshotRequestDao;
     this.snapshotId = snapshotId;
   }
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
-    List<String> authDomains =
-        context
-            .getWorkingMap()
-            .get(SnapshotWorkingMapKeys.SNAPSHOT_AUTH_DOMAIN_GROUPS, new TypeReference<>() {});
-    // Only delete the Sam group if it matches the expected naming pattern
-    var expectedName = IamService.constructSamGroupName(snapshotId.toString());
-    if (Objects.nonNull(authDomains) && authDomains.contains(expectedName)) {
+    // The request will only exist if the snapshot was created with byRequestId mode
+    // If it exists, the request will have the sam group name to be deleted for this snapshot
+    SnapshotAccessRequestModel request = null;
+    try {
+      request = snapshotRequestDao.getByCreatedSnapshotId(snapshotId);
+    } catch (NotFoundException ex) {
+      // If the request does not exist, nothing to delete
```

Comment:
To avoid having to provide an initial value for `request`, and avoid the `if`, you could return early in the `catch`. I think returning early is clearer than relying on local state.
```
    SnapshotAccessRequestModel request;
    try {
      request = snapshotRequestDao.getByCreatedSnapshotId(snapshotId);
    } catch (NotFoundException ex) {
      // If the request does not exist, nothing to delete
      return StepResult.getStepResultSuccess();
    }

    var samGroupName = request.samGroupName();
```